spec.html

```diff
@@ -25228,17 +25228,16 @@ <h1>Runtime Semantics: Decode ( _string_, _reservedSet_ )</h1>
                 1. If the code units at index (_k_ + 1) and (_k_ + 2) within _string_ do not represent hexadecimal digits, throw a *URIError* exception.
                 1. Let _B_ be the 8-bit value represented by the two hexadecimal digits at index (_k_ + 1) and (_k_ + 2).
                 1. Set _k_ to _k_ + 2.
-                1. If the most significant bit in _B_ is 0, then
+                1. Let _n_ be the number of leading 1 bits in _B_.
+                1. If _n_ = 0, then
                   1. Let _C_ be the code unit whose value is _B_.
                   1. If _C_ is not in _reservedSet_, then
                     1. Let _S_ be the String value containing only the code unit _C_.
                   1. Else,
-                    1. Let _S_ be the substring of _string_ from index _start_ to index _k_ inclusive.
+                    1. Let _S_ be the String value equal to the substring of _string_ consisting of the code units at indices _start_ (inclusive) through _k_ (inclusive).
                 1. Else,
-                  1. Assert: the most significant bit in _B_ is 1.
-                  1. Let _n_ be the smallest nonnegative integer such that (_B_ &lt;&lt; _n_) &amp; 0x80 is equal to 0.
-                  1. If _n_ equals 1 or _n_ is greater than 4, throw a *URIError* exception.
-                  1. Let _Octets_ be a List of 8-bit integers of size _n_.
+                  1. If _n_ = 1 or _n_ &gt; 4, throw a *URIError* exception.
+                  1. Let _Octets_ be a List of 8-bit integers of length _n_.
```

Comment:
Spec lists can't have holes, so what are the values in this list? Is it even useful to initially size/populate this list?

Reply:
Good point; refactored to avoid that.